crypto/evp/scrypt.c

```diff
@@ -139,7 +139,7 @@ static void scryptROMix(unsigned char *B, uint64_t r, uint64_t N,
         *pV = *pB++;
         *pV |= *pB++ << 8;
         *pV |= *pB++ << 16;
-        *pV |= *pB++ << 24;
+        *pV |= (uint32_t)*pB++ << 24;
```

Comment:
I always forget that type promotion goes from unsigned to signed.

On "it doesn't affect the result on any of the supported platforms,  because all of them are two's complement" - only if we assume things about the compiler that are not actually safe to assume. I think we should get out of that habit.